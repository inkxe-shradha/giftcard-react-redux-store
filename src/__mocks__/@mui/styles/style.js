import * as Styles from "@mui/styles";

const makeStyles = () => {
  return () => {
    /**
     * Note: if you want to mock this return value to be
     * different within a test suite then use
     * the pattern defined here:
     * https://jestjs.io/docs/en/manual-mocks
     **/
    return {};
  };
};

module.exports = { ...Styles, makeStyles };
