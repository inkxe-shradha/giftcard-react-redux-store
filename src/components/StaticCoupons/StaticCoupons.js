import { Grid, Container, Typography, Button } from "@mui/material";
import { makeStyles } from "@mui/styles";
import React from "react";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import { Link } from "react-router-dom";

const styles = makeStyles((theme) => ({
  paper: {
    padding: theme.spacing(2),
    border: "1px solid" + theme.palette.primary.main,
    wordBreak: "break-all",
  },
  container: {
    marginBlockStart: "1em",
  },
}));
const StaticCoupons = () => {
  const classes = styles();
  return (
    <Container maxWidth="lg" className={classes.container}>
      <Grid container spacing={2}>
        <Grid item xs={12} md={12}>
          <Typography align="center" variant="h4">
            Most Viewed Coupons
          </Typography>
        </Grid>
        <Grid item xs={12} md={12}>
          <Grid container margin="custom" spacing={2}>
            <Grid item xs={12} md={3}>
              <Card>
                <CardMedia
                  component="img"
                  height="140"
                  image="https://pngimg.com/uploads/amazon/amazon_PNG21.png"
                  alt="Amazon Card"
                />
                <CardContent>
                  <Typography gutterBottom variant="h5" component="div">
                    Amazon Gift (5000)
                  </Typography>
                  <Typography variant="body2" color="text.secondary">
                    Amazon Gift Cards are the Perfect Gift, Every Time. Use the
                    eBay Gift Card to shop from millions of items .....
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
            <Grid item xs={12} md={3}>
              <Card>
                <CardMedia
                  component="img"
                  height="140"
                  image="https://i.pinimg.com/originals/15/aa/16/15aa1678d4ee5615c5c53ed5c9968126.png"
                  alt="FlipKart"
                />
                <CardContent>
                  <Typography gutterBottom variant="h5" component="div">
                    Flip-kart (5000)
                  </Typography>
                  <Typography variant="body2" color="text.secondary">
                    Flipkart Pvt Ltd. is an e-commerce company based in
                    Bengaluru, India. Founded by Sachin Bansal and Binny Bansal
                    in 2007 ...
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
            <Grid item xs={12} md={3}>
              <Card>
                <CardMedia
                  component="img"
                  height="140"
                  image="https://img.etimg.com/thumb/msid-52350287,width-300,imgsize-34171,resizemode-4/myntra-app-faces-tech-glitch-spams-users-with-notifications.jpg"
                  alt="Myntra"
                />
                <CardContent>
                  <Typography gutterBottom variant="h5" component="div">
                    Myntra
                  </Typography>
                  <Typography variant="body2" color="text.secondary">
                    Myntra Gift Cards are the Perfect Gift, Every Time. Use the
                    Myntra Gift Card to shop from millions of items in
                    Electronics...
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
            <Grid item xs={12} md={3}>
              <Card>
                <CardMedia
                  component="img"
                  height="140"
                  image="https://cms.qz.com/wp-content/uploads/2015/05/zomatos-4th-logo-2012-14.png?w=900&h=900&crop=1&strip=all&quality=75"
                  alt="Zomato"
                />
                <CardContent>
                  <Typography gutterBottom variant="h5" component="div">
                    Zomato (5000)
                  </Typography>
                  <Typography variant="body2" color="text.secondary">
                    Zomato. Zomato is an Indian restaurant search and discovery
                    service founded in 2008 by Deepinder Goyal ....
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
            {/* View More */}
            <Grid item xs={12} md={12} align="center">
              <Button
                variant="contained"
                color="primary"
                className={classes.button}
                LinkComponent={Link}
                to="/gift-cards"
              >
                View More
              </Button>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Container>
  );
};

export default StaticCoupons;
