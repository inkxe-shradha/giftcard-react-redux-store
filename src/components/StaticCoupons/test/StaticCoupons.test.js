import { createTheme } from "@mui/material";
import { ThemeProvider } from "@mui/styles";
import { render } from "@testing-library/react";
import { mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import StaticCoupons from "../StaticCoupons";

describe("Static coupons description test suite", () => {
  const customRender = (type = "render") => {
    const theme = createTheme();
    if (type === "render") {
      return render(
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <StaticCoupons />
          </BrowserRouter>
        </ThemeProvider>
      );
    } else {
      return mount(
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <StaticCoupons />
          </BrowserRouter>
        </ThemeProvider>
      );
    }
  };
  it("Should create a snapshot of the StaticCoupons  page", () => {
    const { asFragment } = customRender();
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should must have 8 div with the className MuiGrid-grid-md-3", () => {
    const view = customRender("mount");
    expect(view.find(".MuiGrid-grid-md-3").length).toBe(8);
  });

  it("Should have 14 div of className MuiGrid-grid-xs-12", () => {
    const view = customRender("mount");
    expect(view.find(".MuiGrid-grid-xs-12").length).toBe(14);
  });
});
