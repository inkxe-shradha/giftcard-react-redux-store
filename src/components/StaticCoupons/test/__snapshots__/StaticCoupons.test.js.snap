// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static coupons description test suite Should create a snapshot of the StaticCoupons  page 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg makeStyles-container-2 css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-149xwtl-MuiTypography-root"
        >
          Most Viewed Coupons
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-10imocd-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <img
                alt="Amazon Card"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="140"
                src="https://pngimg.com/uploads/amazon/amazon_PNG21.png"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
                >
                  Amazon Gift (5000)
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Amazon Gift Cards are the Perfect Gift, Every Time. Use the eBay Gift Card to shop from millions of items .....
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <img
                alt="FlipKart"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="140"
                src="https://i.pinimg.com/originals/15/aa/16/15aa1678d4ee5615c5c53ed5c9968126.png"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
                >
                  Flip-kart (5000)
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Flipkart Pvt Ltd. is an e-commerce company based in Bengaluru, India. Founded by Sachin Bansal and Binny Bansal in 2007 ...
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <img
                alt="Myntra"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="140"
                src="https://img.etimg.com/thumb/msid-52350287,width-300,imgsize-34171,resizemode-4/myntra-app-faces-tech-glitch-spams-users-with-notifications.jpg"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
                >
                  Myntra
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Myntra Gift Cards are the Perfect Gift, Every Time. Use the Myntra Gift Card to shop from millions of items in Electronics...
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <img
                alt="Zomato"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="140"
                src="https://cms.qz.com/wp-content/uploads/2015/05/zomatos-4th-logo-2012-14.png?w=900&h=900&crop=1&strip=all&quality=75"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
                >
                  Zomato (5000)
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Zomato. Zomato is an Indian restaurant search and discovery service founded in 2008 by Deepinder Goyal ....
                </p>
              </div>
            </div>
          </div>
          <div
            align="center"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
          >
            <a
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
              href="/gift-cards"
              tabindex="0"
            >
              View More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
